server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -275,6 +283,45 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
         Setting.Property.Deprecated
     );
 
+    /**
+     * Sets the cache size for snapshot repository data, defaulting to 1% of node heap memory. The maximum is 5%.
+     */
+    public static final Setting<ByteSizeValue> SNAPSHOT_REPOSITORY_DATA_CACHE_THRESHOLD = new Setting<>(
+        SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME,
+        calculateDefaultSnapshotRepositoryDataCacheThreshold() + "b",
+        (s) -> {
+            ByteSizeValue userDefinedLimit = parseBytesSizeValueOrHeapRatio(s, SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME);
+            long userDefinedLimitBytes = userDefinedLimit.getBytes();
+            long cacheThreshold = calculateDefaultSnapshotRepositoryDataCacheThreshold();
+
+            if (userDefinedLimitBytes > cacheThreshold) {
```

Comment:
We should also limit the lower bound to 500Kb to eliminate any unintended consequences:

```suggestion
            if (userDefinedLimitBytes < CACHE_MIN_THRESHOLD) {
```

Reply:
Wow, thank you so much, @reta!

I have made all the adjustments you mentioned. Thanks to your input, the code quality has improved significantly. I really appreciate it! 😊

I’ve also added a minimum limit and included a test in `MemorySizeSettingsTests` to verify that an error is triggered if the value goes below the minimum. 😄